source/Irrlicht/COpenGLFunctionTable.h

```diff
@@ -0,0 +1,4530 @@
+				IRR_AMD_shader_fragment_mask,
+				IRR_AMD_shader_image_load_store_lod,
+				IRR_AMD_shader_trinary_minmax,
+				IRR_AMD_texture_gather_bias_lod,
+				IRR_NVX_multiview_per_view_attributes,
+				IRR_NV_viewport_array2,
+				IRR_NV_stereo_view_rendering,
+				IRR_NV_sample_mask_override_coverage,
+				IRR_NV_geometry_shader_passthrough,
+				IRR_NV_shader_subgroup_partitioned,
+				IRR_NV_compute_shader_derivatives,
+				IRR_NV_fragment_shader_barycentric,
+				IRR_NV_mesh_shader,
+				IRR_NV_shader_image_footprint,
+				IRR_NV_shading_rate_image,
+				IRR_ARB_shading_language_include,
+				IRR_ARB_shader_stencil_export,
+				IRR_ARB_enhanced_layouts,
+				IRR_ARB_bindless_texture,
+				IRR_ARB_shader_draw_parameters,
+				IRR_ARB_shader_group_vote,
+				IRR_ARB_cull_distance,
+				IRR_ARB_derivative_control,
+				IRR_ARB_shader_texture_image_samples,
+				IRR_KHR_blend_equation_advanced,
+				IRR_KHR_blend_equation_advanced_coherent,
+				IRR_ARB_fragment_shader_interlock,
+				IRR_ARB_gpu_shader_int64,
+				IRR_ARB_post_depth_coverage,
+				IRR_ARB_shader_ballot,
+				IRR_ARB_shader_clock,
+				IRR_ARB_shader_viewport_layer_array,
+				IRR_ARB_sparse_texture2,
+				IRR_ARB_sparse_texture_clamp,
+				IRR_ARB_gl_spirv,
+				IRR_ARB_spirv_extensions,
+				IRR_AMD_shader_stencil_export,
+				IRR_AMD_vertex_shader_viewport_index,
+				IRR_AMD_vertex_shader_layer,
+				IRR_NV_bindless_texture,
+				IRR_NV_shader_atomic_float,
+				IRR_AMD_sparse_texture,
+				IRR_EXT_shader_integer_mix,
+				IRR_INTEL_fragment_shader_ordering,
+				IRR_AMD_shader_stencil_value_export,
+				IRR_NV_shader_thread_group,
+				IRR_NV_shader_thread_shuffle,
+				IRR_EXT_shader_image_load_formatted,
+				IRR_AMD_gpu_shader_int64,
+				IRR_NV_shader_atomic_int64,
+				IRR_EXT_post_depth_coverage,
+				IRR_EXT_sparse_texture2,
+				IRR_NV_fragment_shader_interlock,
+				IRR_NV_sample_locations,
+				IRR_NV_shader_atomic_fp16_vector,
+				IRR_NV_command_list,
+				IRR_OVR_multiview,
+				IRR_OVR_multiview2,
+				IRR_NV_shader_atomic_float64,
+				IRR_INTEL_conservative_rasterization,
+				IRR_NV_conservative_raster_pre_snap,
+				IRR_EXT_shader_framebuffer_fetch,
+				IRR_EXT_shader_framebuffer_fetch_non_coherent,
+				IRR_INTEL_blackhole_render,
+				IRR_NV_shader_texture_footprint,
+				IRR_NV_gpu_shader5
+			};
+			//! show all features with availablity
+			void dump(std::string* outStr = NULL, bool onlyAvailable = false) const;
+
+			void dumpFramebufferFormats() const;
+			bool isDeviceCompatibile(core::vector<std::string>* failedExtensions = NULL);
+
+			static core::CLeakDebugger bufferLeaker;
+			static core::CLeakDebugger textureLeaker;
+			static bool FeatureAvailable[IRR_OpenGL_Feature_Count];
+
+			void initFunctions(bool stencilBuffer);
+			static void loadFunctions();
+			// Some variables for properties
+			bool StencilBuffer;
+			bool TextureCompressionExtension;
+
+			// Some non-boolean properties
+			static int32_t reqUBOAlignment;
```

Comment:
not sure if the class should store anything associated with opengl state's properties

Reply:
Function table only stores function pointers!

Reply:
yeah